theopensystemslab/planx-new
# chore: Insert production data with optional truncation on sync

Rather than clobbering the db and then re-inserting data, I have updated the seed scripts to perform upserts for flows, users and teams.

What this means is that local changes are persisted and new additions are not removed on staging sync or after running `pnpm docker:add-seeds`. This means pizzas and local data can be preserved between syncs. Data can be reset by deleting the record and re-syncing.

This PR should enable send-to-email to work on staging without testing data being overridden (i.e. configured `teams.submission_email` values in staging should be retained after a sync).

**Testing**

This is best tested locally running `pnpm docker:add-seeds` to trigger the sync script:
 - Insert a new team, user and/or flow
 - Amend an existing team, user and/or flow
 - Delete an existing team, user and/or flow
 - Run `pnpm docker:add-seeds` or `PG_USERNAME=${} PG_PASSWORD=${} PG_DATABASE=${} ./scripts/seed-database/sync.sh`
 - Ensure that the new and amended records have persisted
 - Ensure that the deleted record has been restored
